pandas/tseries/frequencies.py

```diff
@@ -265,7 +266,8 @@ def infer_freq(index, warn=True):
                             "type {type}".format(type=type(index)))
         index = index.values
 
-    if not isinstance(index, pd.DatetimeIndex):
+    if not isinstance(index, (DatetimeArrayMixin, TimedeltaArrayMixin,
+                              pd.DatetimeIndex)):
```

Comment:
I'm surprised this hasn't been hit already.  What changed to make this necessary now?

Reply:
This may be unnecessary now. I think it came out of my earlier issues with freq validation. Seeing if it can be removed.